parsers/rhea/rhea.go

```diff
@@ -0,0 +1,742 @@
+	);
+	
+	CREATE TABLE reactivepart (
+		id INT,
+		accession TEXT PRIMARY KEY,
+		name TEXT,
+		htmlname TEXT,
+		compound TEXT NOT NULL REFERENCES compound(accession)
+	);
+	
+	CREATE TABLE reaction (
+		id INT,
+		directional BOOL NOT NULL DEFAULT false,
+		accession TEXT PRIMARY KEY,
+		status TEXT,
+		comment TEXT,
+		equation TEXT,
+		htmlequation TEXT,
+		ischemicallybalanced BOOL NOT NULL DEFAULT true,
+		istransport BOOL NOT NULL DEFAULT false,
+		ec TEXT,
+		location TEXT
+	);
+	
+	CREATE TABLE reactionside (
+		accession TEXT PRIMARY KEY
+	);
+	
+	CREATE TABLE reactionsidereaction (
+		reaction TEXT NOT NULL REFERENCES reaction(accession),
+		reactionside TEXT NOT NULL REFERENCES reactionside(accession),
+		substrateorproduct BOOL NOT NULL DEFAULT false, -- If set to true, disregard substrate below
+	        substrate BOOL NOT NULL DEFAULT false -- If substrateOrProduct is false, "substrate = true" means this is a substrate and "substrate = false" means this is a product
+	);
+	
+	CREATE TABLE reactionparticipant (
+		compound TEXT REFERENCES compound(accession),
+	        reactionside TEXT NOT NULL REFERENCES reactionside(accession),
+	        contains INTEGER,
+	        containsn BOOL NOT NULL DEFAULT false,
+	        minus BOOL NOT NULL DEFAULT false,
+	        plus BOOL NOT NULL DEFAULT false
+	);
+	`
+	_, err := db.Exec(schema)
+	if err != nil {
+		return errors.New("Failed to exec schema. Failed on: " + err.Error())
+	}
+	return nil
+}
+
+// InsertRheaSqlite inserts the Rhea database into an SQLite database with proper normalization for advanced queries.
+func InsertRheaSqlite(db *sql.DB, rhea Rhea) error {
+	// Start transaction with database for insertion. This ensures if there are any problems, they are seamlessly rolled back
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	// First, insert Chebis and Compounds
+	compoundKeys := make(map[string]bool)
+	for _, compound := range rhea.Compounds {
+		// Insert root chebi. Ie, what this current compound's subclass is
+		_, err = tx.Exec("INSERT OR IGNORE INTO chebi(accession) VALUES (?)", compound.SubclassOfChebi)
+		if err != nil {
+			tx.Rollback()
+			return err
+		}
+		// Insert the chebi of the current compound. If it is already inserted, update the subclassification
+		_, err = tx.Exec("INSERT INTO chebi(accession, subclassof) VALUES (?, ?) ON CONFLICT (accession) DO UPDATE SET subclassof = ?", compound.Chebi, compound.SubclassOfChebi, compound.SubclassOfChebi)
+		if err != nil {
+			tx.Rollback()
+			return err
+		}
+		// Insert the compound itself
+		_, err = tx.Exec("INSERT INTO compound(id, accession, position, name, htmlname, formula, charge, chebi, polymerizationindex, compoundtype) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?) ON CONFLICT DO NOTHING", compound.ID, compound.Accession, compound.Position, compound.Name, compound.HTMLName, compound.Formula, compound.Charge, compound.Chebi, compound.PolymerizationIndex, compound.CompoundType)
```

Comment:
Making each of these inserts into a function would increase readability, testability, and make partial updates of the db easier.

Reply:
If the intention was giving users the ability to do incremental or partial updates, I would agree to the functional split. As it stands, the inserts are used precisely once without the intention of letting other people interact with the SQL. Eventually, we will probably want to support more databases (like postgresql) so I think we should table this functionalization until then.